src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineStartService.java

```diff
@@ -0,0 +1,143 @@
+
+		/*
+		PARAMETERS
+		 */
+		IridaWorkflowNamedParameters namedParameters = null;
+
+		if (request.getSavedParameters() != null) {
+			namedParameters = namedParametersService.read(request.getSavedParameters());
+		}
+
+		// Make sure there is a reference files if one is required.
+		if (description.requiresReference() && request.getReference() == null) {
+			throw new ReferenceFileRequiredException(
+					messageSource.getMessage("server.ReferenceFiles.notFound", new Object[] {}, locale));
+		}
+
+		if (request.getAutomatedProjectId() != null) {
+			Project project = projectService.read(request.getAutomatedProjectId());
+			String statusMessage = messageSource.getMessage("analysis.template.status.new", new Object[] {}, locale);
+			AnalysisSubmissionTemplate template = submissionService.createSingleSampleSubmissionTemplate(workflow,
+					request.getReference(), request.getParameters(), namedParameters, request.getName(), statusMessage,
+					request.getDescription(), project, request.isUpdateSamples(), request.sendEmailOnCompletion(),
+					request.sendEmailOnError());
+			return template.getId();
+		} else {
+			/*
+			SHARE RESULTS BACK TO PROJECTS?
+		 	*/
+			List<Project> projects = new ArrayList<>();
+			if (request.getProjects()
+					.size() > 0) {
+				projects = (List<Project>) projectService.readMultiple(request.getProjects());
+			}
+
+			/*
+			SEQUENCE FILES
+		 	*/
+			List<SingleEndSequenceFile> singles = new ArrayList<>();
+			List<SequenceFilePair> pairs = new ArrayList<>();
+			// Check for single ended sequence files
+			Iterable<SequencingObject> sequencingObjects = sequencingObjectService.readMultiple(request.getFileIds());
+			if (description.acceptsSingleSequenceFiles()) {
+				sequencingObjects.forEach(sequencingObject -> {
+					if (sequencingObject instanceof SingleEndSequenceFile) {
+						singles.add((SingleEndSequenceFile) sequencingObject);
+					}
+				});
+			} else if (description.acceptsPairedSequenceFiles()) {
+				sequencingObjects.forEach(sequencingObject -> {
+					if (sequencingObject instanceof SequenceFilePair) {
+						pairs.add((SequenceFilePair) sequencingObject);
+					}
+				});
+			}
+
+			IridaWorkflowInput inputs = description.getInputs();
+			cartService.emptyCart();
+			if (inputs.requiresSingleSample()) {
+				submissionService.createSingleSampleSubmission(workflow, request.getReference(), singles, pairs,
+						request.getParameters(), namedParameters, request.getName(), request.getDescription(), projects,
+						request.isUpdateSamples(), request.sendEmailOnCompletion(), request.sendEmailOnError());
+				return -1L;
+			} else {
+				AnalysisSubmission submission = submissionService.createMultipleSampleSubmission(workflow,
+						request.getReference(), singles, pairs, request.getParameters(), namedParameters,
+						request.getName(), request.getDescription(), projects, request.isUpdateSamples(),
+						request.sendEmailOnCompletion(), request.sendEmailOnCompletion());
```

Comment:
Looks like we have a duplicate passed here `request.sendEmailOnCompletion(), request.sendEmailOnCompletion()`. Should be `request.sendEmailOnCompletion(), request.sendEmailOnError()`

Reply:
Updated in 3b844fbe2bf213f4b34b5ecb091a37995a36ceed - good catch!